module_utils/network_lsr/argument_validator.py

```diff
@@ -353,6 +353,77 @@ def _validate_impl(self, value, name):
         return v
 
 
+class ArgValidatorRange(ArgValidator):
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        name,
+        required=False,
+        val_min=None,
+        val_max=None,
+        default_value=None,
+        numeric_type=int,
+    ):
+        ArgValidator.__init__(self, name, required, default_value)
+        self.val_min = val_min
+        self.val_max = val_max
+        self.numeric_type = numeric_type
+
+    def _validate_num(self, value, name):
+        v = None
+        try:
+            if isinstance(value, bool):
+                # bool can (probably) be converted to self.numeric_type,
+                # but here we don't want to accept a boolean value.
+                pass
+            elif isinstance(value, self.numeric_type):
+                # ArgValidatorNum should normalize the input values to be of type
+                # self.numeric_type, except the default_value
+                v = self.numeric_type(value)
+            else:
+                v2 = self.numeric_type(value)
+                if isinstance(value, Util.STRING_TYPE) or v2 == value:
+                    v = v2
+        except Exception:
+            pass
+        if v is None:
+            raise ValidationError(
+                name, "must be an integer number but is '%s'" % (value)
+            )
+        if self.val_min is not None and v < self.val_min:
+            raise ValidationError(
+                name, "value is %s but cannot be less then %s" % (value, self.val_min)
+            )
+        if self.val_max is not None and v > self.val_max:
+            raise ValidationError(
+                name,
+                "value is %s but cannot be greater then %s" % (value, self.val_max),
+            )
+        return v
+
+    def _validate_impl(self, value, name):
+        match_group = re.match(r"^ *([0-9]+) *- *([0-9]+) *$", value)
+        if match_group:
+            try:
+                val_list = list(map(int, [match_group.group(1), match_group.group(2)]))
```

Comment:
seems hard to read, unless you fully bought into functional programming. And then, why not `list(map(int, map(match_group.group, [1, 2])))` or `list(map(lambda x: int(match_group.group(x)), [1, 2]))` ?

How about `val_list = (int(match_group.group(1)), int(match_group.group(2)))`

Reply:
Updated.